src/logging/index.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as util from 'util';
+import * as logsAPI from '@opentelemetry/api-logs';
+import { OTLPLogExporter } from '@opentelemetry/exporter-logs-otlp-http';
+import { Resource } from '@opentelemetry/resources';
+import { diag } from '@opentelemetry/api';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import {
+  LoggerProvider,
+  BatchLogRecordProcessor,
+  LogRecordProcessor,
+  ConsoleLogRecordExporter,
+} from '@opentelemetry/sdk-logs';
+
+import { getNonEmptyEnvVar, getEnvArray, defaultServiceName } from '../utils';
+import { detect as detectResource } from '../resource';
+
+type LogRecordProcessorFactory = (
+  options: LoggingOptions
+) => LogRecordProcessor | LogRecordProcessor[];
+
+interface LoggingOptions {
+  accessToken?: string;
+  realm?: string;
+  serviceName: string;
+  endpoint?: string;
+  resource: Resource;
+  logRecordProcessorFactory: LogRecordProcessorFactory;
+}
+
+export const allowedLoggingOptions = [
+  'accessToken',
+  'realm',
+  'serviceName',
+  'endpoint',
+  'logRecordProcessorFactory',
+];
+
+export type StartLoggingOptions = Partial<Omit<LoggingOptions, 'resource'>>;
+
+export function startLogging(opts: StartLoggingOptions = {}) {
+  const options = _setDefaultOptions(opts);
+  const loggerProvider = new LoggerProvider({
+    resource: options.resource,
+  });
+
+  let processors = options.logRecordProcessorFactory(options);
+
+  if (!Array.isArray(processors)) {
+    processors = [processors];
+  }
+
+  processors.forEach((processor) =>
+    loggerProvider.addLogRecordProcessor(processor)
+  );
+
+  logsAPI.logs.setGlobalLoggerProvider(loggerProvider);
+
+  return {
+    stop: () => {
+      return loggerProvider.shutdown();
+    },
+  };
+}
+
+export function _setDefaultOptions(
+  options: StartLoggingOptions = {}
+): LoggingOptions {
+  let resource = detectResource();
+
+  const serviceName =
+    options.serviceName ||
+    getNonEmptyEnvVar('OTEL_SERVICE_NAME') ||
+    resource.attributes[SemanticResourceAttributes.SERVICE_NAME];
+
+  if (!serviceName) {
+    diag.warn(
+      'service.name attribute is not set, your service is unnamed and will be difficult to identify. ' +
```

Comment:
Might want to specify that it is from the logging part, else we'll get 2 identical warnings (1 from tracing)